src/core/raster/qgsrastershader.cpp

```diff
@@ -169,6 +170,7 @@ void QgsRasterShader::readXml( const QDomElement& elem )
   if ( !colorRampShaderElem.isNull() )
   {
     QgsColorRampShader* colorRampShader = new QgsColorRampShader();
+    colorRampShader->setColorRampName( colorRampShaderElem.attribute( "colorRampName", "" ) );
```

Comment:
And also here the default does [not need to explicitly written](http://doc.qt.io/qt-4.8/qdomelement.html#attribute).

I.e., this is just fine

colorRampShader->setColorRampName( colorRampShaderElem.attribute( "colorRampName" ) );